src/Subscriber/FlowSubscriber.php

```diff
@@ -82,8 +82,8 @@ private function sendMail(DataBag $dataBag, string $mailTemplateTypeId, Context
             $dataBag->set('contentPlain', $plain);
         }
 
-        if ($subject) {
-            $salesChannelId = $dataBag->getString('salesChannelId');
+        $salesChannelId = $dataBag->get('salesChannelId') ?: null;
```

Comment:
you could still call getString, and in the if look if it's not empty string

Reply:
No get string throws an exception if salesChannelId is not set
```
Uncaught PHP Exception Symfony\Component\HttpFoundation\Exception\UnexpectedValueException: "Parameter value "salesChannelId" cannot be converted to "string"." at ParameterBag.php line 132 {"exception":"[object] (Symfony\\Component\\HttpFoundation\\Exception\\UnexpectedValueException(code: 0): Parameter value \"salesChannelId\" cannot be converted to \"string\". at /var/www/.../vendor/symfony/http-foundation/ParameterBag.php:132)"} []
```

Reply:
I also tried that. has seems to return true, although salesChannelId cannot be converted to "string".